spotbugs-tests/src/test/java/edu/umd/cs/findbugs/TextUICommandLineTest.java

```diff
@@ -78,4 +80,28 @@ void htmlReportWithOption() throws IOException {
         assertTrue(xdocsFile.toFile().isFile());
         assertTrue(textFile.toFile().isFile());
     }
+
+    @Test
+    void sharedOutputFilesAreNotDuplicated() throws Exception {
```

Comment:
Can you please add a test where not the same path is added, but two paths refer to the same location? E.g. the first path is `tempFile.xml` and the second one is `tempDir/../tempFile.xml` (obviously without calling `getAbsolutePath()` or canonizing the paths in the test.)

Reply:
What would be the expected behavior for that case? The original intention of the Bugreport was to make this use only a single outputstream, i.e. only a single reporter, which would imply canonicalizing the path in the underlying code...

Reply:
IMO the same as if the same file is added, so that, it is not duplicated.

Reply:
Added test case `deduplicationUsesCanonicalPaths`